services/scan/src/precinct_scanner_app.ts

```diff
@@ -14,43 +14,85 @@ import express, { Application } from 'express';
 import { readFile } from 'fs-extra';
 import multer from 'multer';
 import { z } from 'zod';
+import { interpretTemplate } from '@votingworks/ballot-interpreter-vx';
 import { PrecinctScannerStateMachine } from './precinct_scanner_state_machine';
 import { pdfToImages } from './util/pdf_to_images';
 import { Workspace } from './util/workspace';
-import { createInterpreter, loadLayouts } from './simple_interpreter';
 import { backup } from './backup';
+import { PrecinctScannerInterpreter } from './precinct_scanner_interpreter';
+import { Store } from './store';
 
 function sum(nums: number[]) {
   return nums.reduce((a, b) => a + b, 0);
 }
 
 type NoParams = never;
 
-async function configureMachine(
-  machine: PrecinctScannerStateMachine,
+/**
+ * Loads ballot layouts from the {@link Store} to be used by the interpreter.
+ * The results may be cached and used again as long as the underlying HMPB
+ * templates are not modified.
+ */
+async function loadLayouts(
+  store: Store
+): Promise<BallotPageLayoutWithImage[] | undefined> {
+  const electionDefinition = store.getElectionDefinition();
+  if (!electionDefinition) return;
+
+  const templates = store.getHmpbTemplates();
+  const loadedLayouts: BallotPageLayoutWithImage[] = [];
+
+  for (const [pdf, layouts] of templates) {
+    for await (const { page, pageNumber } of pdfToImages(pdf, { scale: 2 })) {
+      const ballotPageLayout = layouts[pageNumber - 1];
+      loadedLayouts.push(
+        await interpretTemplate({
+          electionDefinition,
+          imageData: page,
+          metadata: ballotPageLayout.metadata,
+        })
+      );
+    }
+  }
+
+  return loadedLayouts;
+}
+
+/**
+ * Loads all of the relevant configuration from the workspace store and
+ * configures the interpreter. Should be called anytime one of these config
+ * values is changed in order to update the interpreter's config.
+ */
+async function configureInterpreter(
+  interpreter: PrecinctScannerInterpreter,
   workspace: Workspace
 ) {
   const { store } = workspace;
   const electionDefinition = store.getElectionDefinition();
   assert(electionDefinition);
   const layouts = await loadLayouts(store);
   assert(layouts);
-  const interpreter = createInterpreter({
+  interpreter.configure({
     electionDefinition,
     ballotImagesPath: workspace.ballotImagesPath,
     testMode: store.getTestMode(),
     markThresholdOverrides: store.getMarkThresholdOverrides(),
     layouts,
   });
-
-  machine.configure(store, interpreter);
 }
 
-export function buildPrecinctScannerApp(
+export async function buildPrecinctScannerApp(
   machine: PrecinctScannerStateMachine,
+  interpreter: PrecinctScannerInterpreter,
   workspace: Workspace
-): Application {
+): Promise<Application> {
   const { store } = workspace;
+
+  // Load interpreter configuration from the store on startup
+  if (store.getElectionDefinition()) {
+    await configureInterpreter(interpreter, workspace);
```

Comment:
@eventualbuddha suggested a while back that we don't `void` this promise in case it errors